privacyidea/lib/utils/__init__.py

```diff
@@ -1077,20 +1077,12 @@ def censor_connect_string(connect_string):
     """
     Take a SQLAlchemy connect string and return a sanitized version
     that can be written to the log without disclosing the password.
-    The password is replaced with "xxxx".
+    The password is replaced with "***".
     In case any error occurs, return "<error when censoring connect string>"
     """
     try:
-        parsed = urlparse(connect_string)
-        if parsed.password is not None:
-            # We need to censor the ``netloc`` attribute: user:pass@host
-            _, host = parsed.netloc.rsplit("@", 1)
-            new_netloc = u'{}:{}@{}'.format(parsed.username, 'xxxx', host)
-            # Convert the URL to six components. netloc is component #1.
-            splitted = list(parsed)
-            splitted[1] = new_netloc
-            return urlunparse(splitted)
-        return connect_string
+        parsed = sqlalchemy.engine.url.make_url(connect_string)
+        return repr(parsed)
```

Comment:
it seems that in python2 `repr()` does not only call the `__repr__()` function of the object. In this case it breaks the tests if the username does contain non-ASCII characters.
Could You change this to `return parsed.__repr__()` which does work in Python 2.7.

Reply:
Hi @plettich thanks for reviewing this PR. Of course I will change it. I'm sorry I only tested Python 3. Good catch.